docs/api-guide.md

```diff
@@ -654,6 +654,14 @@ You can't specify clustering-key boundaries and orderings in `Scan` without spec
 
 `Put` is an operation to put a record specified by a primary key. The operation behaves as an upsert operation for a record, in which the operation updates the record if the record exists or inserts the record if the record does not exist.
 
+{% capture notice--info %}
+**Note**
+
+When you update an existing record, you need to read the record by using `Get` or `Scan` before using a `Put` operation. Otherwise, the operation will fail due to a conflict. This occurs because ScalarDB assumes that another transaction wrote the record. Instead of reading the record explicitly, you can enable implicit pre-read. See [Enable implicit pre-read for `Put` operations](#enable-implicit-pre-read-for-put-operations) for details.
```

Comment:
I think your understanding is correct. However, since this document is not intended to explain the details of the transaction mechanism in ScalarDB, we might not need to explain it deeply here. Or do you have an idea for this? If yes, can you please suggest it? Thanks.

Reply:
It might be a bit deep (so it might not be proper for this document), but how about the following description?

```suggestion
When you update an existing record, you need to read the record by using `Get` or `Scan` before using a `Put` operation. Otherwise, the operation will fail due to a conflict. ScalarDB needs to check whether another transaction wrote the record for concurrency control when it runs write operations, but ScalarDB cannot do that if you don't read the existing record before write operations. This causes the conflict error. Instead of reading the record explicitly, you can enable implicit pre-read. See [Enable implicit pre-read for `Put` operations](#enable-implicit-pre-read-for-put-operations) for details.
```

---

Or, how about the following simple description? I think this is not a deep.

```suggestion
When you update an existing record, you need to read the record by using `Get` or `Scan` before using a `Put` operation. Otherwise, the operation will fail due to a conflict. This occurs because the specification of ScalarDB to manage transactions properly. Instead of reading the record explicitly, you can enable implicit pre-read. See [Enable implicit pre-read for `Put` operations](#enable-implicit-pre-read-for-put-operations) for details.
```

Reply:
@komamitsu (CC: @feeblefakie) What do you think about @kota2and3kan's idea?

Reply:
I noticed [Enable implicit pre-read for `Put` operations](#enable-implicit-pre-read-for-put-operations) already has a detailed description about the background. So, the following simple one is okay?

```
When you update an existing record, you need to read the record by using `Get` or `Scan` before using a `Put` operation. The implicit pre-read feature would be useful for that. See [Enable implicit pre-read for `Put` operations](#enable-implicit-pre-read-for-put-operations) for details.
```

Reply:
Okay, thank you. Let me fix with `This occurs because of the specification of ScalarDB to manage transactions properly.` Thanks.

Reply:
> Only one thing:
> 
> > This occurs because the specification of ScalarDB to manage transactions properly.
> 
> `This occurs because of the specification of ScalarDB to manage transactions properly.` seems a bit better.

Ah, sorry... I overlooked the grammar error... 🙇 
I fixed my suggestion just in case!

```suggestion
When you update an existing record, you need to read the record by using `Get` or `Scan` before using a `Put` operation. Otherwise, the operation will fail due to a conflict. This occurs because of the specification of ScalarDB to manage transactions properly. Instead of reading the record explicitly, you can enable implicit pre-read. See [Enable implicit pre-read for `Put` operations](#enable-implicit-pre-read-for-put-operations) for details.
```

Reply:
Fixed in dfec35af2614d6d0b67a3a0d88f7c6b0a16fb2fe. Thanks.